sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/higherOrderFunctions.scala

```diff
@@ -445,8 +445,64 @@ case class ArrayExists(
       false
     }
   }
+}
+
+/**
+ * Tests whether a predicate holds for all elements in the array.
+ */
+@ExpressionDescription(usage =
+  "_FUNC_(expr, pred) - Tests whether a predicate holds for all elements in the array.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_(array(1, 2, 3), x -> x % 2 == 0);
+       false
+      > SELECT _FUNC_(array(2, 4, 8), x -> x % 2 == 0);
+       true
+      > SELECT _FUNC_(array(1, null, 3), x -> x % 2 == 0);
+       NULL
```

Comment:
> if the right-hand array contains any null elements and no false comparison result is obtained, the result of ALL will be null, not true

```
presto> SELECT 1 > ALL (VALUES 2, null, 3);
 _col0 
-------
 false 
(1 row)
```

The result of last example isn't false?

Reply:
yeah, it should be `false`.

For example,

```
SELECT _FUNC_(array(2, null, 8), x -> x % 2 == 0);
```
should be `null`.